Nodejs/Product/Nodejs/TypeScriptHelpers/TsConfigJsonFactory.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.NodejsTools.TypeScript
+{
+    internal static class TsConfigJsonFactory
+    {
+        public static async Task<TsConfigJson> CreateAsync(string fullPathToFile)
+        {
+            if (fullPathToFile == null || !fullPathToFile.EndsWith(NodejsConstants.TsConfigJsonFile, StringComparison.OrdinalIgnoreCase))
+            {
+                throw new ArgumentException("Expected full path to 'tsconfig.json' file.", nameof(fullPathToFile));
+            }
+
+            if (File.Exists(fullPathToFile))
+            {
+                var retryInterval = 500;
+                var attempts = 5;
+
+                // populate _source with retries for recoverable errors.
+                while (--attempts >= 0)
+                {
+                    try
+                    {
+                        using (var fin = new FileStream(fullPathToFile, FileMode.Open, FileAccess.Read, FileShare.Read))
+                        using (var reader = new StreamReader(fin))
+                        {
+                            object source = null;
+
+                            var text = await reader.ReadToEndAsync();
```

Comment:
>await reader.ReadToEndAsync() [](start = 39, length = 29)

Doing it this way because `File.ReadAllText` is synchronous?

Reply:
yes